types/index.d.ts

```diff
@@ -15,6 +15,7 @@ import type { getFundingSources } from "./components/funding-eligibility";
 
 export interface PayPalNamespace {
     Buttons?: (options?: PayPalButtonsComponentProps) => PayPalButtonsComponent;
+    HostedFields?: (options?: PayPalButtonsComponentProps) => PayPalButtonsComponent;
```

Comment:
HostedFields returns a PayPalButtonsComponent?

Reply:
Seeing now that you are having a conversation with @gregjopa about this in `react-paypal-js`.

Reply:
Let's create a new `PayPalHostedFieldsComponent` since this is a different interface than Buttons.
```suggestion
    HostedFields?: PayPalHostedFieldsComponent;
```
<img width="267" alt="hosted-fields-screenshot" src="https://user-images.githubusercontent.com/534034/112323446-ca6fec80-8c7f-11eb-95c8-a6061590f8aa.png">

I'm thinking we can create a new file here: types/components/hosted-fields.d.ts. And it will contain the following:
```ts
export interface PayPalHostedFieldsComponent {
    isEligible: () => boolean;
    render: (container: HTMLElement | string) => Promise<void>;
}
```

Reply:
How did it change those to Buttons, I swear I changed it to 
`HostedFields?: (options?: PayPalHostedFieldsComponentProps) => PayPalHostedFieldsComponent;`

It's not a function that returns `PayPalHostedFieldsComponent` like buttons?

Reply:
> It's not a function that returns PayPalHostedFieldsComponent like buttons?

Unfortunately the hosted-fields component does not follow the same pattern. Instead it returns an object. The source code for this components lives here: https://github.com/paypal/paypal-card-components/blob/6d099d7157d1ba26cfe2f2e0858c6ec24450a20d/src/component.js#L152-L168

Reply:
Ohhhhhhh, so you're saying the type isn't specific to the react components, but the JS SDK , but couldn't we extend it for react then?